pkg/internal/client/txn.go

```diff
@@ -706,6 +707,11 @@ func (txn *Txn) send(ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.
 	// If we're not yet writing in this txn, but intend to, insert a
 	// begin transaction request before the first write command.
 	if needBeginTxn {
+		// If we're setting the system-config trigger, anchor the transaction to
```

Comment:
This doesn't seem like the right place to put this logic. The logic around system-config is too specific for `internal/client`. Instead, the code which creates the transaction to modify system-config should make sure the first key modified is the system config span. This is similar to what we do when we split or merge ranges.